frontend/src/components/test-history.js

```diff
@@ -33,334 +31,317 @@ import {
-      });
-  }
+    if (JSON.stringify(filtersState) !== '{}') {
+      let historyFilters = {...filtersState};
+      // disregard result filter (we want all results)
+      delete historyFilters['result'];
+      let api_filter = toAPIFilter(historyFilters).join();
 
-  getResultsForTable () {
-    if (this.state.comparisonResults !== undefined) {
-      this.setState({
-        results: this.state.comparisonResults,
-        rows: this.state.comparisonResults.map((result, index) => resultToTestHistoryRow(result, index, this.setFilter)).flat(),
+      let summary = Object.assign({
+        'passes': 0,
+        'failures': 0,
+        'errors': 0,
+        'skips': 0,
+        'xfailures': 0,
+        'xpasses': 0
       });
-    } else {
-      const filters = this.state.filters;
-      this.setState({rows: [getSpinnerRow(4)], isEmpty: false, isError: false});
-      let params = buildParams(filters);
-      params['filter'] = toAPIFilter(filters);
-      params['pageSize'] = this.state.pageSize;
-      params['page'] = this.state.page;
-      params['estimate'] = 'true';
 
-      this.setState({rows: [['Loading...', '', '', '', '']]});
-      HttpClient.get([Settings.serverUrl, 'result'], params)
+      HttpClient.get(
+        [Settings.serverUrl, 'widget', 'result-aggregator'],
+        {
+          group_field: 'result',
+          additional_filters: api_filter,
+        }
+      )
         .then(response => HttpClient.handleResponse(response))
-        .then(data => this.setState({
-          results: data.results,
-          rows: data.results.map((result, index) => resultToTestHistoryRow(result, index, this.setFilter)).flat(),
-          page: data.pagination.page,
-          pageSize: data.pagination.pageSize,
-          totalItems: data.pagination.totalItems,
-          totalPages: data.pagination.totalPages,
-          isEmpty: data.pagination.totalItems === 0,
-        }))
-        .catch((error) => {
-          console.error('Error fetching result data:', error);
-          this.setState({rows: [], isEmpty: false, isError: true});
+        .then(data => {
+          data.forEach(item => {
+            summary[RESULT_STATES[item['_id']]] = item['count'];
+          });
+          setHistorySummary(summary);
         });
     }
+  }, [filtersState]);
+
+  const onFailuresCheck = (checked) => {
+    setFiltersState({
+      ...filtersState,
+      'result': {
+        ...filtersState['result'],
+        'val': ('failed;error') + ((checked) ? ';skipped;xfailed' : ';skipped;xfailed;xpassed;passed')
+      }
+    });
+    setOnlyFailures(checked);
+  };
 
-  }
+  const onDropdownSelect  = (_event, selection) => {
+    if (testResult?.start_time) {
+      const startTime = new Date(testResult?.start_time);
+      // here a selection (month) is considered to be 30 days, and there are 86400*1000 ms in a day
+      let timeRange = new Date(startTime.getTime() - (selection * 30 * 86400 * 1000));
```

Comment:
Nitpick: Could be const since we are not re-assigning object but changing it's props